defs.bzl

```diff
@@ -34,6 +34,12 @@ def _pip_repository_impl(rctx):
         "@%s" % rctx.attr.name,
     ]
 
+    if rctx.attr.extra_pip_args:
+        args += [
+            "--extra_pip_args",
+            "\"" + " ".join(rctx.attr.extra_pip_args) + "\"",
```

Comment:
It would be good to confirm the behavior here if an argument contains a quote. i.e `"\"foo\""`. Can we confirm that the escaping of such a quote is kept when repacked into a string.

Reply:
I think the key is in `extract_wheels/__init__.py`. It uses `strip`, whose definition says: "Remove spaces at the beginning and at the end of the string" and parameters: "A set of characters to remove as leading/trailing characters". We add quotes at the beginning and the end and then remove those. At least it means we keep the string intact.

I also tried it, with providing index urls quoted as:

```
extra_pip_args = [
    "--extra-index-url",
    "\"https://some-uri/abc/def/\"",
],
```

It worked.

Reply:
I also printed args in `extract_wheels/__init__.py`.

```[..., '--extra-index-url', '"https://some-uri/abc/def/"', ...]```